acados/ocp_qp/ocp_qp_hpipm.c

```diff
@@ -323,6 +323,28 @@ void ocp_qp_hpipm_memory_reset(void *config_, void *qp_in_, void *qp_out_, void
     ocp_qp_hpipm_memory_assign(config_, qp_in->dim, opts_, mem_);
 }
 
+void ocp_qp_hpipm_solver_get(void *config_, void *qp_in_, void *qp_out_, void *opts_, void *mem_, const char *field, int stage, void* value)
```

Comment:
This would work as expected only if there is no partial condensing and no x0 elimination.
E.g. some possible issues:
- in case of partial condensing, it could be stage > N of qp_in   =>   random result or segfault
- in case of partial condensing, it could be nu[stage] in qp_in > nu NLP that is used in ocp_nlp_interface to tell the expected sizes   =>   memory corruption or segfault
- in case of x0 elimination, there would be no K[0] or P[0] returned at all in the value buffer

Reply:
I addressed this here. But only in Python https://github.com/acados/acados/pull/975/commits/6c37ff53bb7c288917dfc4e40652589da946b5d4

Reply:
Can you also add a comment in the C interface to warn that this would not work in general, and maybe as a TODO that someone can fix it in the future?

Reply:
Actually an easy solution would be to pass two additional arguments to the solver_get function, namely int size1, int size2. These are set in the ocp_nlp_interface.c using the NLP dim there, and compared with the dim in ocp_qp_in. For stage it has to be stage < or <= the N in the ocp_qp_in dims, depending on the field needed.

Reply:
Done as suggested in https://github.com/acados/acados/pull/975/commits/15b74362f7231824f7caaa7b391fb3cca2b12f61